### DISCUSSION TITLE
MySQL on Galera

### DISCUSSION THREAD

**Discussion by solracsf** [juicedata/juicefs#1544] [General]
Clusters are complicated because of atomicity of transactions; but can Galera `wrep_sync_wait` help here?

Either by setting it globally server side (`SET GLOBAL wsrep_sync_wait = 1` ([or 7](https://mariadb.com/kb/en/galera-cluster-system-variables/#wsrep_sync_wait))), or by issue these command by `juicefs` internals directly when galera is detected?

`SET SESSION wsrep_sync_wait=1; SELECT ...; SET SESSION wsrep_sync_wait=0;`

See https://mariadb.com/kb/en/galera-cluster-system-variables/#wsrep_sync_wait

**Comment by davies** [comment]
@solracsf It seems that the server side configuration will be better, so we don't need to change JuiceFS for a special deployment.

In the other way, we could use this trick to implement the transaction for MariaDB in the MySQL driver other than in JuiceFS.
